www/nginx/src/opnsense/mvc/app/views/OPNsense/Nginx/tabbed_dialog.volt

```diff
@@ -0,0 +1,183 @@
+ # INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ # CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ # POSSIBILITY OF SUCH DAMAGE.
+ #}
+
+{#
+ # Generate input dialog, uses the following parameters (as associative array):
+ #
+ # fields          :   list of field type objects, see form_input_tr tag for details
+ # id              :   form id, used as unique id for this modal form. inner form to place data is called frm_[id]
+ #                     save button is identified by btn_[id]_save
+ # label           :   dialog label
+ #}
+
+{# Volt templates in php7 have issues with scope sometimes, copy input values to make them more unique #}
+{% set base_dialog_id=id %}
+{% set base_dialog_fields=fields %}
+{% set base_dialog_label=label %}
+
+{# Find if there are help supported or advanced field on this page #}
+{% set base_dialog_help=false %}
+{% set base_dialog_advanced=false %}
+{% for field in base_dialog_fields|default({})%}
+    {% for name,element in field %}
+        {% if name=='help' %}
+            {% set base_dialog_help=true %}
+        {% endif %}
+        {% if name=='advanced' %}
+            {% set base_dialog_advanced=true %}
+        {% endif %}
+    {% endfor %}
+    {% if base_dialog_help|default(false) and base_dialog_advanced|default(false) %}
+        {% break %}
+    {% endif %}
+{% endfor %}
+
+<script>
+    $(document).ready(function() {
+        // hook into on-show event to extend validation
+        $('#{{base_dialog_id}}').on('shown.bs.modal', function (e) {
+            $('.nav-tabs a[href="#frm_{{base_dialog_id}}-tab_general"]').tab('show');
+
+            $("#btn_{{base_dialog_id}}_save").click(function() {
+                // TODO: Search for tab with validation errors, but currently only the first tab has even validation rules
+                $('.nav-tabs a[href="#frm_{{base_dialog_id}}-tab_general"]').tab('show');
+            });
+        })
+
+        // Read currently selected tab in main form and store for later
+        $('#{{base_dialog_id}}').on('show.bs.modal', function (e) {
+            $('#{{base_dialog_id}}').attr("data-inittab", window.location.hash);
+        });
+        // Restore selected tab of main form as URL was modified by dialog tabs
+        $('#{{base_dialog_id}}').on('hide.bs.modal', function (e) {
+            window.location.hash = $('#{{base_dialog_id}}').attr("data-inittab");
+        });
+    });
+</script>
+
+<div class="modal fade" id="{{base_dialog_id}}" tabindex="-1" role="dialog" aria-labelledby="{{base_dialog_id}}Label" aria-hidden="true">
+    <div class="modal-backdrop fade in"></div>
+    <div class="modal-dialog modal-lg">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
```

Comment:
not sure what thsi aria-label does but it might need to be translated.

Reply:
You might be right, but that is a copy and paste from core's layout_partials...

https://github.com/opnsense/core/blob/a5f1e2c7c4b8f2802809dcccaee8821926668104/src/opnsense/mvc/app/views/layout_partials/base_dialog.volt#L63

I'd suggest to keep it as is for now. Agree?

Reply:
https://github.com/opnsense/core/commit/a7ad2845a17 :)

Reply:
@8191 Since @fichtner pushed the change to core I guess this is a sign that it should be changed here as well